source/guides/clear/swupd-3rd-party.rst

```diff
@@ -0,0 +1,304 @@
+
+   In this example, the `helloclear.sh` app simply prints
+   "Hello Clear!" when invoked.  
+
+   .. code-block:: bash
+
+      # make helloclear directory
+      mkdir -p helloclear/usr/bin && pushd $_
+
+      # create helloclear.sh script
+      cat > helloclear.sh << EOF
+      #!/bin/bash
+      echo "Hello Clear!"
+      EOF
+
+      # make script executable
+      chmod +x helloclear.sh
+
+      popd
+
+Create bundle of 3rd-party app with mixer
+*****************************************
+
+Next, use :command:`mixer` to create a bundle for each of the apps from the previous section.  
+
+#. Install the mixer tool.
+   
+   .. code-block:: bash
+
+      sudo swupd bundle-add mixer
+
+#. Create a mixer workspace.
+   
+   .. code-block:: bash
+
+      mkdir ~/mixer && cd $_
+
+#. Initialize a mix without any default bundles.
+   
+   .. code-block:: bash
+
+      mixer init --no-default-bundles
+
+#. Configure :file:`builder.conf` to set the default bundle, CONTENTURL, and VERSIONURL.  
+   For the "URL"s in this example, it will be IP address of the web server that was
+   set up earlier.
+   Substitute <IP-address-of-web-server> with the IP address of your host.  
+
+   .. code-block:: bash
+      
+      mixer config set Swupd.BUNDLE "os-core"
+      mixer config set Swupd.CONTENTURL "http://<IP-address-of-web-server>"
+      mixer config set Swupd.VERSIONURL "http://<IP-address-of-web-server>"
+
+#. Create an empty local `os-core` bundle.  :command:`swupd` client expects the 
+   `os-core` bundle to exist in a mix even if it’s empty.
+
+   .. code-block:: bash
+      
+      mixer bundle create os-core --local
+
+#. Using the `helloclear.sh` app as an example, create the `helloclear` bundle 
+   and use the `content()` directive with the path to the `helloclear` directory.
+   
+   .. code-block:: bash
+      
+      mixer bundle create helloclear
+      echo "content($HOME/my-3rd-party-apps/helloclear/)" > local-bundles/helloclear
+
+#. Add both bundles to the mix.
+
+   .. code-block:: bash
+      
+      mixer bundle add os-core
+      mixer bundle add helloclear
+
+#. Build the bundles and generate the update content.
+
+   .. code-block:: bash
+      
+      sudo mixer build bundles
+      sudo mixer build update
+
+Install and manage 3rd-party bundle on client system
+****************************************************
+
+Finally, use the :command:`swupd` client tool to install and 
+manage the bundle created with :command:`mixer` earlier.
+All installed 3rd-party bundle reside in :file:`/opt/3rd-party`.
+
+#. First, add a repo link to the web server.
+   The `os-core` bundle will be added automatically when adding a repo.  
+   It contains items that mixer injects into the mix such as version information, 
+   format, CONTENTURL, VERSIONURL, and certificate.
+    
+   .. code-block:: bash
+
+      sudo swupd 3rd-party add my-3rd-party-repo \
+      http://<IP-address-of-web-server> --allow-insecure-http
```

Comment:
In running these commands, I receive the error:
`
Error: Impossible to import 3rd party repository certificate
To ignore certificate check use --nosigcheck
`
BTW: I already set up the /etc/swupd/config file --last year.

Reply:
I don't seem to have any trouble.  see my screenshot. 

<img width="721" alt="Screen Shot 2020-03-18 at 7 11 24 PM" src="https://user-images.githubusercontent.com/29260389/77024682-5ef88e00-694c-11ea-8ce8-70767fcfada6.png">

Reply:
This is it: "Error: Impossible to import 3rd party repository certificate"
See screenshot below.  

Let's put aside this error for further testing; I'll create a separate issue for this and assign it to you in our repo.  This error does not block merging of this PR; it's difficult to reproduce. 

![image](https://user-images.githubusercontent.com/37549381/77113040-51dcad00-69e7-11ea-954e-ad251cc5af7f.png)